---
title: "Phylogenetic Comparative Analysis Package and Vignette"
output: rmarkdown::html_vignette
date: "`r Sys.Date()`"
vignette: >
  %\VignetteIndexEntry{Phylogenetic Comparative Analysis Package and Vignette}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---
*Collaborators: Savvy Cornett & Jyhreh Johnson*

## Background
Phylogenetic Comparative Analysis or Phylogenetic Comparative Methods (PCMs) are a set of statistical methods used by biologist, geneticist, and anthropologists to analyze phylogenetic trees and track/present species traits (Pennell & Harmon, 2013).\

## Introduction
In this package, we have included a dataset for the Sonic Hedgehog vertebrate orthologs (SHH_orthologs.csv) from the NIH database. We also have two functions: one for making a phylogenetic tree, and the other for doing a phylogenetic scatterplot.\

## Preliminaries
Load in the necessary libraries for the package. 
```{r Preliminaries, message=FALSE, warning=FALSE}
require(devtools) # to make package
require(usethis) # to make package
require(roxygen2) # to make package
require(withr) # to make package
# for our package
require(tidyverse)
require(ape) #phylo
require(ggplot2)
require(phangorn)
require(phytools) 
require(geomorph)
require(BiocManager) # to get packages from bioconductor (ie package not on CRAN)
BiocManager::install("ggtree") # install ggTree package. will give two prompts, answer "all" and "yes" to them respectively
require(ggtree) # extension of ggplot to make phylo trees
BiocManager::install("msa") # install "msa" package.  will give two prompts, answer "all" and "yes" to them respectively
require(msa) # multiple sequence alignment
```
## Viewing the data

For this project, we have included an example dataset to use with our PCMs package, *SHH_orthologs_Carnivora*, #explain the dataset 
```{r}
#load in the data
f_name <- "https://raw.githubusercontent.com/jyhrehjohnson/cornett-johnson_ADA_GroupProject/main/SHH_orthologs_Carnivora.csv"
#"https://raw.githubusercontent.com/jyhrehjohnson/cornett-johnson_ADA_GroupProject/main/SHH_orthologs.csv" (switched out for the above)

#use the function to read in the data and create the r dataset
sonic <- read_file(f_name)
```
## Running PCMs

## Plotting PCMs
```{r}
#input plot function and discuss plotting ^^
```
## Interpreting PCMs Plots

## References\ 
Pennell, M. W., & Harmon, L. J. (2013). An integrative view of phylogenetic comparative methods: Connections to population genetics, community ecology, and paleobiology: Integrative comparative methods. *Annals of the New York Academy of Sciences, 1289*(1), 90–105. <https://doi.org/10.1111/nyas.12157>\

Toparslan, E., Karabag, K., & Bilge, U. (2020). A workflow with R: Phylogenetic analyses and visualizations using mitochondrial cytochrome b gene sequences. *PLOS ONE, 15*(12), e0243927. <https://doi.org/10.1371/journal.pone.0243927>\

Yu, G., Smith, D. K., Zhu, H., Guan, Y., & Lam, T. T. (2017). ggtree: An r package for visualization and annotation of phylogenetic trees with their covariates and other associated data. *Methods in Ecology and Evolution, 8*(1), 28–36. <https://doi.org/10.1111/2041-210X.12628>\
